varnish/README.md

```diff
@@ -1,54 +1,593 @@
 
-Install the `dd-check-varnish` package manually or with your favorite configuration manager
+# Installation
 
-## Configuration
+The varnish check is packaged with the Agent, so simply [install the Agent](https://app.datadoghq.com/account/settings#agent) on your varnish servers. If you need the newest version of the check, install the `dd-check-varnish` package.
 
-Configure the Agent to connect to Varnish
+# Configuration
+
+If you're running Varnish 4.1+, add the dd-agent system user to the varnish group (e.g. `sudo usermod -G varnish -a dd-agent`).
+
+Then, create a file `varnish.yaml` in the Agent's `conf.d` directory:
 
- - Edit conf.d/varnish.yaml
 ```
 init_config:
 
 instances:
-    - varnishstat: /usr/bin/varnishstat
-      tags:
-          - instance:production
+  - varnishstat: /usr/bin/varnishstat        # or wherever varnishstat lives
+    varnishadm: <PATH_TO_VARNISHADM_BIN>     # to submit service checks for the health of each backend
+#   secretfile: <PATH_TO_VARNISH_SECRETFILE> # if you configured varnishadm and your secret file isn't /etc/varnish/secret
+#   tags:
+#     - instance:production
 ```
 
- - If you're running Varnish 4.1+, you must add the dd-agent user to the varnish group.
+If you don't set `varnishadm`, the Agent won't check backend health. If you do set it, the Agent needs privileges to execute the binary with root privileges. Add the following to your `/etc/sudoers` file:
+
+```
+dd-agent ALL=(ALL) NOPASSWD:/usr/bin/varnishadm
 ```
-sudo usermod -a -G varnish dd-agent
+
+Restart the Agent to start sending varnish metrics and service checks to Datadog.
+
+# Validation
+
+Run the Agent's `info` subcommand and look for `varnish` under the Checks section:
+
 ```
+  Checks
+  ======
+    [...]
+
+    varnish
+    -------
+      - instance #0 [OK]
+      - Collected 26 metrics, 0 events & 1 service check
+
+    [...]
+```
+# Compatibility
+
+The Varnish check is compatible with all major platforms.
+
+# Service Checks
+
+**varnish.backend_healthy**:
+
+The Agent submits this service check if you configure `varnishadm`. It submits a service check for each varnish backend, tagging each with `backend:<backend_name>`.
 
- - If you want the check to use `varnishadm` and send a service check, the agent must be able to access `varnishadm` with root privileges. For this, you can update your `/etc/sudoers` file with for example:
- ```
- dd-agent ALL=(ALL) NOPASSWD:/usr/bin/varnishadm
- ```
+# Further Reading
 
+See our [series of blog posts](https://www.datadoghq.com/blog/top-varnish-performance-metrics/) about monitoring varnish with Datadog.
 
+# Metrics
 
-## Validation
+See [metadata.csv](https://github.com/DataDog/integrations-core/blob/master/varnish/metadata.csv) for a list of metrics provided by this check.
 
-When you run `datadog-agent info` you should see something like the following:
+The check submits different metrics depending on the major version of varnish:
 
-    Checks
-    ======
+### Varnish 3.x
```

Comment:
@hush-hush Users like to see these by-version lists of metrics. I've seen many support tickets about it for other checks (e.g. "Which metrics does the Elasticsearch check send if I'm on version x.y?")

I just spent >5 minutes looking through Varnish's docs and couldn't find a straightforward list like we've got here. I'd rather not throw away information that we already have that can help some users and prevent them from clicking around in Varnish's docs—and not getting answers—like I just did.

Having said that, this list is unwieldy and is not the best way to present things. I just copied it over quickly from Docs site so the information wouldn't get lost. This information really belongs in the metrics metadata, as I see it. 

(by the way, the fact that the metrics "may change in the futur and the names are taken from varnish itself" would seem to be an argument against curating metrics metadata, too. Some Agent checks *do* avoid listing some metrics in metadata if their names tend to be "dynamic". But for this check we did create metadata.)

What do you think about adding this to metrics metadata? e.g.

`varnish.bans_lurker_tested,gauge,,object,second,Bans and objects tested against each other by the ban-lurker.,0,varnish,bans lurker tested`

becomes

`varnish.bans_lurker_tested,gauge,,object,second,Bans and objects tested against each other by the ban-lurker; Only reported by varnish 4.x and later.,0,varnish,bans lurker tested`